### DISCUSSION TITLE
Adding support to gds for enums not represented by I32

### DISCUSSION THREAD

**Discussion by SMorettini** [nasa/fprime#2187] [Ideas]
Hello,

At present, fprime-gds operates under the assumption that every enum is represented as an I32, as detailed in the documentation: [Link to Documentation](https://fprime-community.github.io/fpp/fpp-users-guide.html#Defining-Enums_The-Representation-Type).

I am interested in extending fprime-gds to accommodate various representation types for enums. Could this proposed feature find acceptance within the Fprime community?

If the response to the aforementioned query is affirmative, could you kindly guide me on the appropriate method for updating the dictionary to encompass this new information?

My recommendation involves incorporating a new field named `serialize_type` within the enum.

**Example**:
```
    <enum type="MyType" serialize_type="U8">
      <item name="first_item" value="0" description=""/>
      ...
    </enum>
```

**Comment by JackNWhite** [comment]
I also have my eye on doing this, but, for completeness:
https://github.com/fprime-community/fprime-tools/issues/17
